### PR TITLE
Added precompiled container build instructions for DOCA drivers

### PR DESCRIPTION
**Pull request by e0ne**

### CODE REVIEW

File: docs/advanced-configurations.rst [hunk 1]
```diff
@@ -304,3 +304,118 @@ If self-signed certificates are used for an HTTPS based internal repository, a C
...
+-------------
+
+Before you begin, ensure that you have the following prerequisites:
+
+~~~~~~
+Common
+~~~~~~
+
+- Docker (Ubuntu) / Podman (RH) installed on your build system.
+- Web access to NVIDIA NIC drivers sources. Latest NIC drivers published at `NIC drivers download center <https://network.nvidia.com/products/infiniband-drivers/linux/mlnx_ofed/>`_, for example: `https://www.mellanox.com/downloads/ofed/MLNX_OFED-24.01-0.3.3.1/MLNX_OFED_SRC-debian-24.01-0.3.3.1.tgz <https://www.mellanox.com/downloads/ofed/MLNX_OFED-24.01-0.3.3.1/MLNX_OFED_SRC-debian-24.01-0.3.3.1.tgz>`_
```
Comment (krembu) [inline docs/advanced-configurations.rst:323]: should take last GA

File: docs/advanced-configurations.rst [hunk 2]
```diff
@@ -304,3 +304,118 @@ If self-signed certificates are used for an HTTPS based internal repository, a C
...
+To build RHEL-based image please use provided :download:`Dockerfile <files/RHEL_Dockerfile>` and :download:`entrypoint script <files/entrypoint.sh>`.
+
+.. code-block:: bash
+
+   podman build \
+    --build-arg D_OS=rhel9.2 \
+    --build-arg D_ARCH=x86_64 \
+    --build-arg D_KERNEL_VER=5.14.0-284.32.1.el9_2.x86_64 \
+    --build-arg D_OFED_VERSION=24.01-0.3.3.1 \
+    --build-arg D_BASE_IMAGE="registry.redhat.io/openshift4/driver-toolkit-rhel9:v4.13.0-202309112001.p0.gd719bdc.assembly.stream" \
```
Comment (krembu) [inline docs/advanced-configurations.rst:395]: D_BASE_IMAGE is the DTK image should explain above to use this param for OCP4.14^ and how to detect required DTK image

File: docs/advanced-configurations.rst [hunk 3]
```diff
@@ -304,3 +304,118 @@ If self-signed certificates are used for an HTTPS based internal repository, a C
...
+    --build-arg D_FINAL_BASE_IMAGE=registry.access.redhat.com/ubi9/ubi:latest \
+    --tag 24.01-0.3.3.1-0-5.14.0-284.32.1.el9_2-rhel9.2-amd64 \
```
Comment (krembu) [inline docs/advanced-configurations.rst:397]: version

File: docs/files/RHEL_Dockerfile [hunk 4]
```diff
@@ -0,0 +1,142 @@
+# Common (multistage) args
+ARG D_OS="rhel9.2"
+ARG D_ARCH="x86_64"
+ARG D_CONTAINER_VER="0"
+ARG D_OFED_VERSION="23.10-0.5.5.0"
```
Comment (krembu) [inline docs/files/RHEL_Dockerfile:5]: version

File: docs/files/RHEL_Dockerfile [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+
+ARG D_PYTHON_VERSION="36"
+ARG D_PYTHON="python${D_PYTHON_VERSION}"
+
+FROM $D_BASE_IMAGE AS base
+
+# Inherited global args
+ARG D_OS
+
+# https://redmine.mellanox.com/issues/3528150/issue_history#note-9
```
Comment (krembu) [inline docs/files/RHEL_Dockerfile:28]: clean?

File: docs/advanced-configurations.rst (hunk 1 above)
Comment (e0ne) [reply docs/advanced-configurations.rst:323]: done

File: docs/files/RHEL_Dockerfile (hunk 5 above)
Comment (e0ne) [reply docs/files/RHEL_Dockerfile:28]: done

File: docs/advanced-configurations.rst (hunk 3 above)
Comment (e0ne) [reply docs/advanced-configurations.rst:397]: done

File: docs/files/RHEL_Dockerfile (hunk 4 above)
Comment (e0ne) [reply docs/files/RHEL_Dockerfile:5]: done

File: docs/files/RHEL_Dockerfile [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+    autoconf python3-devel ethtool automake pciutils libtool hostname
+
+RUN set -x && \
+# Download NVIDIA NIC driver sources
+    mkdir -p ${D_OFED_SRC_DOWNLOAD_PATH} && \
+    cd ${D_OFED_SRC_DOWNLOAD_PATH} && (curl -sL ${D_OFED_URL_PATH} | tar -xzf -)
+
+WORKDIR /
+ADD ./entrypoint.sh /root/entrypoint.sh
+ADD ./dtk_nic_driver_build.sh /root/dtk_nic_driver_build.sh
```
[the remark this answers is not in the document]
Comment (e0ne) [reply docs/files/RHEL_Dockerfile:74]: fixed

File: docs/advanced-configurations.rst (hunk 2 above)
Comment (e0ne) [reply docs/advanced-configurations.rst:395]: done

**Review by rollandf — APPROVED** [review]

**Pull request merged**